### ISSUE TITLE
Color contrast is insufficient - the "error count" button in the dashboard

### ISSUE THREAD

**Issue by AmyJuneH** [backdrop/backdrop-issues#6119]
Color contrast is insufficient - the "error count" button in the dashboard

The "error count" button in the dashboard has insufficient color contrast.
"The color of the text and the color of the background are not in sufficient contrast to each other.
The contrast ratio should be at least 4.5:1 for normal text and 3:1 for large text.
Note: Contrasts are not relevant where content is hidden. But make sure that for hidden content, which becomes visible when receiving focus, the contrast ratio is sufficient.
Insufficient contrast between text and its background can give problems for users with low vision and color blindness, but it can also affect a lot of other users, leading to important text potentially being overlooked."

Results:
3.39 out of a required 4.5
 #eeeeee / rgb(238,238,238)
Background color
 #ee3d23 / rgb(238,61,35)

**Comment by kiamlaluno** [comment]
I cannot find an *error count* button on the dashboard. Where is it placed? (I take it is always visible.)

**Comment by laryn** [comment]
I believe @AmyJuneH is referring to the error indicator in the admin bar:
<img width="1024" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/871421/9b860251-3d9e-49d7-94e7-04680cea28c0">

**Comment by AmyJuneH** [comment]
Yes Thank you @laryn

**Comment by wesruv** [comment]
PR is ready, I fixed a lot of contrast and hover/focus a11y issues (including the one in the ticket):
https://github.com/backdrop/backdrop/pull/4804

**Comment by laryn** [comment]
I just updated the branch with the latest core commits so the tests will run again.

**Comment by klonos** [comment]
Thanks @wesruv 🙏🏼 ...I need to take a better look, but as initial feedback:
- I like the "underline" (`box-shadow`) in the top-level menu items that are in the active trail 👍🏼 ...although I'd prefer the inverse: blue for the current, and white for the hovered-over. The reason being that as it is currently, the white underline "blends" with the rest of the page, and it gives the appearance of the menu item being "elevated" or mispositioned by an accidental margin/padding or something. Side-by-side screenshot/mockup of what I mean:
  ![image](https://github.com/backdrop/backdrop-issues/assets/2423362/c7ebdc98-2785-4aa2-95e5-21c3c2c1e658)
- The underline/shadow is nice, but I would prefer the active trail (and hover?) to be a more obvious color difference, rather than the subtle shades of black. Now that we have SVG icons that can be adapted to the font color, I believe that we could be more "brave" and do some white-on-black/black-on-white (some shades of these anyway). Something like this:
  <img width="1157" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/78cb7e2c-cd13-424e-b897-e5e8091bb4e0">
- I like the color/contrast changes on the error badge 👍🏼 (although not so much the addition of the shadow)

**Comment by wesruv** [comment]
I messed around with it further, I like the active item being white with a blue border, got that added.

I noticed the focus state wasn't being caught by the CSS when navigating with keyboard, so I addressed that, and added a 2px outline of the blue color to give a halo to the focused item. That feels more obvious and looks nice.

**Comment by laryn** [comment]
Keyboard accessibility might be a bigger issue on the admin menu (I can't seem to get the submenus to activate with a keyboard, and the "This page" item gets skipped over completely as shown in the screencast below). The search box also could get a focus outline.
![CleanShot 2024-06-27 at 18 52 44](https://github.com/backdrop/backdrop-issues/assets/871421/c5e5deb5-ca4a-4784-8fff-3d9d044a1d73)

**Comment by wesruv** [comment]
@laryn yea, I'm noticing that too. The site is still usable without sub menus, and I think I'd rather change the tech behind the menu than try to fix what's there. I don't think we have any menu JS in Backdrop land that's up to snuff.

Good catch on the search! I'll get that added.

**Comment by wesruv** [comment]
Hrmm, @izmeez I'm not able to reproduce that bug, I double checked, and I _shouldn't_ have changed any layout CSS. I wonder if it was a caching issue or something weird happening?

![image](https://github.com/user-attachments/assets/6ea09606-ce39-4f8e-ab7c-7ac2e3aa7569)

Can anyone else reproduce that issue?

**Comment by avpaderno** [comment]
I saw the preview site for this PR as @wesruv sees it. 
I recall I have seen something like @izmeez screenshot in another preview site or a site.

**Comment by avpaderno** [comment]
I can reproduce it: Changing the browser's window size, when there is not enough space to the right of the menu item, the sub-menu is rendered over the existing menu item, as in @izmeez screenshot.

I do not recall this is specific for this PR. If I remember correctly, it happened also with a different preview site.  
I will check what happens with a preview site for a different, maybe newer, PR.

**Comment by avpaderno** [comment]
I tried with the preview site for https://github.com/backdrop/backdrop/pull/4818.  
When there is not enough space to the right of the menu item, the sub-menu item is drawn over the menu item, but it completely covers it. I can clearly read the sub-menu item text.

**Comment by klonos** [comment]
Thanks for the work on this @wesruv 🙏🏼 ...this looks like a considerable improvement over the current situation 👍🏼 

> I noticed the focus state wasn't being caught by the CSS when navigating with keyboard, so I addressed that, and added a 2px outline of the blue color to give a halo to the focused item. That feels more obvious and looks nice.

That halo looks nice indeed, but for the "extra" items in the toolbar, it also has a white outline in addition to the blue one you've added:
<img width="514" alt="image" src="https://github.com/user-attachments/assets/69ed5c28-8f30-4ef8-b0ff-c7803c21f2a4">

> Keyboard accessibility might be a bigger issue on the admin menu (I can't seem to get the submenus to activate with a keyboard, and the "This page" item gets skipped over completely as shown in the screencast below). That's not part of this issue.

I've noticed the same problems while testing this @laryn, and indeed not part of the scope of this issue here.

> ... I think I'd rather change the tech behind the menu than try to fix what's there. I don't think we have any menu JS in Backdrop land that's up to snuff.

@wesruv we have #2809 for the a11y and keyboard navigation, and now #6669 for the tab index for the "This page" item 😉 

> Can anyone else reproduce that issue?

Yes @wesruv, I can reproduce it, same as @avpaderno in the previous comments here. On a vanilla installation of Backdrop, on a demo sandbox (without the changes in the PR provided here), the items expand towards the opposite direction, but the top-most item has a background that prevents the text of any underlaying item to blend with the text of the expanded item:
<img width="406" alt="image" src="https://github.com/user-attachments/assets/8dc5b21c-46f4-43b4-a39e-4f4b97f7cc2f">
...in the PR sandbox, the background is missing, which makes the expanded item become "see-through", and causes the text overlap.

To reproduce, you'll need to reduce the width of your window to a point where the "extra" items are collapsed/hidden away, and only the main navigation items are left. Try to reduce the with to the point just before the main navigation items also collapse. Then, hover over the previous-to-last top-menu item (Configuration), and try hovering over any menu item that goes 3 levels deep (like the "Date and time formats" item that @izmeez mentioned).

Other feedback:
- I really like the removal of the underline-on-hover for the text of the menu items 👍🏼 ...it's been annoying me for a while, so thanks 🙏🏼 
- I really like the blue background color used when hovering over items via the search bar:
<img width="877" alt="image" src="https://github.com/user-attachments/assets/012f93e5-d1eb-4ef3-aa2f-dd6c79127b8f">
  Can we please use this same blue color for hover-over for all non-top-level items?
- Speaking of hover-over color/indication, any menu item that does not have any children does not get one as it is now.

The fixes in the PR have expanded to more than the original bug report here, which warrants a title/scope change I think. I know that there's many people that don't like when that happens, but I'm fine with it in this case here. I really want to see this overhaul get into the next bug fix release.

**Comment by wesruv** [comment]
Created a follow up PR for the issue @izmeez brought up since it's distinct from this one: https://github.com/backdrop/backdrop-issues/issues/6687

**Comment by wesruv** [comment]
Created a follow up issue for what @laryn noticed: https://github.com/backdrop/backdrop-issues/issues/6688

There some some significant a11y issues there.

> Keyboard accessibility might be a bigger issue on the admin menu (I can't seem to get the submenus to activate with a keyboard, and the "This page" item gets skipped over completely as shown in the screencast below). That's not part of this issue.
> 
> As far as this issue goes, the search box also could get a focus outline.

**Comment by stpaultim** [comment]
@wesruv - I am a bit confused about if there are additional issues mentioned above that still need to be addressed OR if you have fixed everything you plan to fix in this issue AND moved over problems to other issues?

Does this issue need more testing/feedback now?

**Comment by jenlampton** [comment]
I really like the new red color, and the blue highlight. I think it's a nice improvement, and I'd love to see this change in conjunction with https://github.com/backdrop/backdrop-issues/issues/6710. Adding bug-fix milestone candidate label.
